MykSuite/src/main/java/com/infomaniak/lib/myksuite/ui/screens/MyKSuiteUpgradeBottomSheet.kt

```diff
@@ -0,0 +1,143 @@
+    style: ButtonType,
+    onButtonClicked: () -> Unit,
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .verticalScroll(rememberScrollState()),
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+
+        val paddedModifier = Modifier.padding(horizontal = Margin.Large)
+        Image(
+            imageVector = ImageVector.vectorResource(R.drawable.illu_banner),
+            contentScale = ContentScale.FillWidth,
+            contentDescription = null,
+            modifier = Modifier.fillMaxWidth(),
+        )
+        Text(
+            modifier = paddedModifier,
+            text = stringResource(R.string.myKSuiteUpgradeTitle),
+            textAlign = TextAlign.Center,
+            style = MyKSuiteTheme.typography.h2,
+            color = MyKSuiteTheme.colors.primaryTextColor,
+        )
+        Spacer(Modifier.height(Margin.Medium))
+        Text(
+            modifier = paddedModifier,
+            text = stringResource(R.string.myKSuiteUpgradeDescription),
+            style = MyKSuiteTheme.typography.bodyRegular,
+            color = MyKSuiteTheme.colors.secondaryTextColor,
+        )
+        Spacer(Modifier.height(Margin.Medium))
+        UpgradeFeatures(customFeatures, paddedModifier)
+        Text(
+            modifier = paddedModifier,
+            text = stringResource(R.string.myKSuiteUpgradeDetails),
+            style = MyKSuiteTheme.typography.bodyRegular,
+            color = MyKSuiteTheme.colors.secondaryTextColor,
+        )
+        Spacer(Modifier.height(Margin.Huge))
+        Button(
+            modifier = paddedModifier
+                .fillMaxWidth()
+                .height(Dimens.buttonHeight),
+            colors = style.colors().buttonColors(),
+            shape = style.shape,
+            onClick = onButtonClicked,
+        ) {
+            Text(stringResource(R.string.buttonClose))
+        }
+        Spacer(Modifier.height(Margin.Large))
+    }
+}
+
+@Composable
+private fun ColumnScope.UpgradeFeatures(
+    customFeatures: @Composable (() -> List<MyKSuiteUpgradeFeatures>)?,
+    modifier: Modifier,
+) {
+    customFeatures?.let {
+        it().forEach { UpgradeFeature(it, modifier) }
+        UpgradeFeature(MyKSuiteUpgradeFeatures.MoreFeatures, modifier)
+        Spacer(Modifier.height(Margin.Large))
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Preview(name = "(1) Light")
+@Preview(name = "(2) Dark", uiMode = Configuration.UI_MODE_NIGHT_YES or Configuration.UI_MODE_TYPE_NORMAL)
+@Composable
+private fun Preview() {
+    MyKSuiteTheme {
+        Surface {
+            MyKSuiteUpgradeBottomSheet(
+                onDismissRequest = {},
+                style = ButtonType.Mail,
+                customFeatures = {
+                    listOf(MyKSuiteUpgradeFeatures(title = R.string.myKSuiteUpgradeLabel, icon = R.drawable.ic_gift))
+                },
+            )
+        }
+    }
+}
```

Comment:
Usually it's better to have the BottomSheetContent previewed here. But in your specific case I can also see the need to use the interactive mode with the actual bottom sheet. But maybe we can have two previews, one with the sheet and one with the content?

Reply:
Also if we keep the MyKSuiteUpgradeBottomSheet preview. Right now the MyKSuiteTheme is defined once inside MyKSuiteUpgradeBottomSheet and a second time inside the Preview